python/thunder/rdds/series.py

```diff
@@ -636,6 +636,111 @@ def query(self, inds, var='inds', order='F', onebased=True):
+        sorted and no records have been removed by filtering, then the resulting output should be equivalent
+        to what one would get from calling myImages.saveAsBinarySeries().
+
+        The json config file written out by this method takes the value of the 'nvalues' key from the
+        length of this object's .index attribute. If the index has been modified to have a different length
+        than the number of values in a Series record, then the resulting data files will be read back in
+        incorrectly.
+
+        Parameters
+        ----------
+        outputdirname : string path or URI to directory to be created
+            Output files will be written underneath outputdirname. This directory must not yet exist
+            (unless overwrite is True), and must be no more than one level beneath an existing directory.
+            It will be created as a result of this call.
+
+        overwrite : bool
+            If true, outputdirname and all its contents will be deleted and recreated as part
+            of this call.
+        """
+        import cStringIO as StringIO
+        import struct
+        from thunder.rdds.imgblocks.blocks import SimpleBlocks
+        from thunder.rdds.fileio.writers import getParallelWriterForPath
+        from thunder.rdds.fileio.seriesloader import writeSeriesConfig
+
+        def partitionToBinarySeries(kvIter):
+            """Collects all Series records in a partition into a single binary series record.
+            """
+            keypacker = None
+            firstKey = None
+            buf = StringIO.StringIO()
+            for seriesKey, series in kvIter:
+                if keypacker is None:
+                    keypacker = struct.Struct('h'*len(seriesKey))
+                    firstKey = seriesKey
+                # print >> sys.stderr, seriesKey, series, series.tostring().encode('hex')
+                buf.write(keypacker.pack(*seriesKey))
+                buf.write(series.tostring())
+            val = buf.getvalue()
+            buf.close()
+            # we might have an empty partition, in which case firstKey will still be None
+            if firstKey is None:
+                return iter([])
+            else:
+                label = SimpleBlocks.getBinarySeriesNameForKey(firstKey) + ".bin"
+                return iter([(label, val)])
+
+        writer = getParallelWriterForPath(outputdirname)(outputdirname, overwrite=overwrite)
+
+        binseriesrdd = self.rdd.mapPartitions(partitionToBinarySeries)
+
+        binseriesrdd.foreach(writer.writerFcn)
+        # note that we use len(self.index) here for nvalues! we don't really have a great alternative
+        # to this at the moment, short of calling first(). 
+        writeSeriesConfig(outputdirname, len(self.dims), len(self.index), dims=self.dims.count,
+                          keytype='int16', valuetype=self.dtype, overwrite=overwrite)
```

Comment:
One practical consequence of using `int16` for the keys is that the resulting files can basically only be read by us. You cannot really load one of these things in, e.g., numpy or matlab, because those can't handle non-constant numerical type specs (as far as I know). We might not need to fix that... but if we wanted to, we could update the keys to the type of the values before writing. Thoughts?